core/domain/blog_validators.py

```diff
@@ -0,0 +1,123 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Validators for Blog models. """
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import base_model_validators
+from core.platform import models
+
+datastore_services = models.Registry.import_datastore_services()
+
+(base_models, blog_models) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.blog
+    ])
+
+
+class BlogPostModelValidator(base_model_validators.BaseModelValidator):
+    """Class for validating BlogPostModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        # Valid id: random_hash of 12 chars.
+        regex_string = '[A-Za-z0-9-_]{1,%s}$' % (base_models.ID_LENGTH)
+        return regex_string
```

Comment:
```suggestion
        return '[A-Za-z0-9-_]{1,%s}$' % (base_models.ID_LENGTH)
```

Optional nit^ (ditto in other places)

Reply:
Done